torch/_library/utils.py

```diff
@@ -78,24 +78,29 @@ def is_functional_schema(schema: Any) -> bool:
     - it has at least one return
     """
 
+    def is_functional(schema):
+        if schema.is_mutable:
+            return False
+        rets = schema.returns
+        is_non_mutating_view = len(rets) > 0 and any(
+            r.alias_info is not None and not r.alias_info.is_write for r in rets
```

Comment:
How could "is_write" be true here since we check for is_mutable False above?

Reply:
is_mutable only checks that the inputs are written to :/
All this code is pre-existing, I am not changing the logic. I agree that it's messy.